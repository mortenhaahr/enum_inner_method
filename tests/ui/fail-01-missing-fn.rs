use enum_inner_method::enum_inner_method;

pub struct Circle;
impl Circle {
   // Note: Doesn't implement area
}

pub struct Rectangle;
impl Rectangle {
    pub fn area(&self, x: f64) -> f64 {
        6.0 * x
    }
}

#[enum_inner_method(area(f64) -> f64)]
pub enum ShapeEnum {
    Circle(Circle),
    Rectangle(Rectangle),
}

fn main() {
}
