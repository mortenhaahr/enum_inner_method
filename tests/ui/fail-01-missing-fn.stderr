error[E0599]: no method named `area` found for reference `&Circle` in the current scope
  --> tests/ui/fail-01-missing-fn.rs:15:21
   |
15 | #[enum_inner_method(area(f64) -> f64)]
   |                     ^^^^ method not found in `&Circle`
